src/Type/Constant/ConstantArrayType.php

```diff
@@ -386,6 +390,17 @@ public function findTypeAndMethodName(): ?ConstantArrayTypeAndMethod
 			if ($this->isOptionalKey(0) || $this->isOptionalKey(1)) {
 				$has = $has->and(TrinaryLogic::createMaybe());
 			}
+			if ($scope === null) {
+				$scope = new OutOfClassScope();
+			}
+			if (
+				$isClassString
+				&& $has->yes()
+				&& !$scope->isInClass()
+				&& !$type->getMethod($method->getValue(), $scope)->isStatic()
+			) {
+				$has = $has->and(TrinaryLogic::createMaybe());
+			}
 
```

Comment:
So whenever checking if something `isCallable` the scope is always out fo class scope, which is what causes `CallStaticMethodsRuleTest::testBug1971` to fail. 

Now I'm realizing that I messed up this logic a bit for not having inClass and checking the class name, as well.

```
				if (
					!$method->isStatic()
					&& $scope->isInClass()
					&& $scope->getClassReflection()->getName() !== $classReflection->getName()
					&& !$scope->getClassReflection()->isSubclassOf($classReflection->getName())
				) {
```

Going to step away from it and clear my head.